src/activity_handlers.cpp

```diff
@@ -536,23 +536,34 @@ void activity_handlers::hotwire_finish( player_activity *act, player *)
 
 void activity_handlers::longsalvage_finish( player_activity *act, player *p )
 {
-    bool has_salvage_tool = p->has_items_with_quality( "CUT", 1, 1 );
-    if( !has_salvage_tool ) {
-        add_msg(m_bad, _("You no longer have the necessary tools to keep salvaging!"));
-    }
-
+    item &salvage_tool = p->i_at( act->values[2] );
```

Comment:
Actually, nevermind, looks like @Tivec and I managed to figure it out after all. Do take a look at #11524 and make sure it's still sane, though.